In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline

import warnings
warnings.filterwarnings("ignore")

In [7]:
df=pd.read_excel("SALESDATA.xlsx")

In [8]:
df.head()

CustKey    DateKey  Discount Amount Invoice Date  Invoice Number  \
0  10000481.0 2017-04-30          -237.91   2017-04-30        100012.0   
1  10002220.0 2017-07-14           368.79   2017-07-14        100233.0   
2  10002220.0 2017-10-17           109.73   2017-10-17        116165.0   
3  10002489.0 2017-06-03          -211.75   2017-06-03        100096.0   
4  10004516.0 2017-05-27         96627.94   2017-05-27        103341.0   

  Item Class Item Number                             Item  Line Number  \
0        NaN         NaN                 Urban Large Eggs       2000.0   
1        P01     20910.0               Moms Sliced Turkey       1000.0   
2        P01     38076.0  Cutting Edge Foot-Long Hot Dogs       1000.0   
3        NaN         NaN                         Kiwi Lox       1000.0   
4        P01     60776.0             High Top Sweet Onion       1000.0   

   List Price  Order Number Promised Delivery Date  Sales Amount  \
0        0.00      200015.0             2017-04-30        237.91   
1      824.96      200245.0             2017-07-14        456.17   
2      548.66      213157.0             2017-10-16        438.93   
3        0.00      200107.0             2017-06-03        211.75   
4      408.52      203785.0             2017-05-28      89248.66   

   Sales Amount Based on List Price  Sales Cost Amount  Sales Margin Amount  \
0                              0.00                0.0               237.91   
1                            824.96                0.0               456.17   
2                            548.66                0.0               438.93   
3                              0.00                0.0               211.75   
4                         185876.60                0.0             89248.66   

   Sales Price  Sales Quantity  Sales Rep U/M  
0   237.910000             1.0      184.0  EA  
1   456.170000             1.0      127.0  EA  
2   438.930000             1.0      127.0  EA  
3   211.750000             1.0      160.0  EA  
4   196.150901           455.0      124.0  SE

In [9]:
#Checking for shape
df.shape

(65282, 20)

Observations: The data contain total 65282 rows and 20 columns

In [11]:
#Displacing the Column Names
df.columns

Index(['CustKey', 'DateKey', 'Discount Amount', 'Invoice Date',
       'Invoice Number', 'Item Class', 'Item Number', 'Item', 'Line Number',
       'List Price', 'Order Number', 'Promised Delivery Date', 'Sales Amount',
       'Sales Amount Based on List Price', 'Sales Cost Amount',
       'Sales Margin Amount', 'Sales Price', 'Sales Quantity', 'Sales Rep',
       'U/M'],
      dtype='object')

In [12]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 65282 entries, 0 to 65281
Data columns (total 20 columns):
 #   Column                            Non-Null Count  Dtype         
---  ------                            --------------  -----         
 0   CustKey                           65282 non-null  float64       
 1   DateKey                           65282 non-null  datetime64[ns]
 2   Discount Amount                   65280 non-null  float64       
 3   Invoice Date                      65282 non-null  datetime64[ns]
 4   Invoice Number                    65282 non-null  float64       
 5   Item Class                        56993 non-null  object        
 6   Item Number                       65241 non-null  object        
 7   Item                              65282 non-null  object        
 8   Line Number                       65282 non-null  float64       
 9   List Price                        65282 non-null  float64       
 10  Order Number                      65282 non-nu

In [14]:
#Checking for null Values
df.isnull().sum()

CustKey                                0
DateKey                                0
Discount Amount                        2
Invoice Date                           0
Invoice Number                         0
Item Class                          8289
Item Number                           41
Item                                   0
Line Number                            0
List Price                             0
Order Number                           0
Promised Delivery Date                 0
Sales Amount                           0
Sales Amount Based on List Price       0
Sales Cost Amount                      0
Sales Margin Amount                    0
Sales Price                            1
Sales Quantity                         0
Sales Rep                              0
U/M                                    0
dtype: int64

Observation:
    There are 2 null values in Discount Amount Columns, 8289 null values in Item Class, 41 null values in Item Number and 1 null value in Sales Price.

In [15]:
df1= df.copy()

In [17]:
df1["Item Class"].value_counts()

P01    56977
PO1       16
Name: Item Class, dtype: int64

In [19]:
#Getting the descrptive data 
df1.describe().round(2)

CustKey  Discount Amount  Invoice Number  Line Number  List Price  \
count     65282.00         65280.00        65282.00     65282.00    65282.00   
mean   10017702.71          1855.57       216223.66     23713.85      514.69   
std        7176.15          9037.14        94992.28     32664.02      449.19   
min    10000453.00       -255820.80       100012.00      1000.00        0.00   
25%    10012715.00           246.04       117931.00      3000.00      181.56   
50%    10019665.00           441.76       222869.50     12000.00      325.19   
75%    10023511.00           999.76       314318.75     32000.00      803.86   
max    10027583.00        343532.66       332842.00    344000.00     2760.70   

       Order Number  Sales Amount  Sales Amount Based on List Price  \
count      65282.00      65282.00                          65282.00   
mean      180583.06       2852.04                           4707.47   
std        67593.87      15164.34                          20696.44   
min       100838.00        200.01                              0.00   
25%       115321.00        308.39                            561.04   
50%       203702.00        553.94                            998.16   
75%       218576.00       1280.04                           2315.04   
max       321532.00     555376.00                         632610.16   

       Sales Cost Amount  Sales Margin Amount  Sales Price  Sales Quantity  \
count           65282.00             65282.00     65281.00        65282.00   
mean             1660.98              1191.06       283.62           45.08   
std              9556.49              5860.79       252.88          429.66   
min                 0.00             -3932.93     -5000.00           -1.00   
25%               167.79               129.95       100.07            2.00   
50%               304.50               246.49       183.76            3.00   
75%               687.32               579.53       448.22            8.00   
max            366576.00            188800.00      6035.00        16000.00   

       Sales Rep  
count   65282.00  
mean      137.42  
std        26.64  
min       103.00  
25%       113.00  
50%       134.00  
75%       160.00  
max       185.00

In [20]:
#By using the Invoice date column creating the date, Month and year columns

df1["Invoice_year"]=df["Invoice Date"].dt.year
df1["Invoice_Month"]=df["Invoice Date"].dt.month
df1["Invoice_Date"]=df["Invoice Date"].dt.date

In [21]:
df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 65282 entries, 0 to 65281
Data columns (total 23 columns):
 #   Column                            Non-Null Count  Dtype         
---  ------                            --------------  -----         
 0   CustKey                           65282 non-null  float64       
 1   DateKey                           65282 non-null  datetime64[ns]
 2   Discount Amount                   65280 non-null  float64       
 3   Invoice Date                      65282 non-null  datetime64[ns]
 4   Invoice Number                    65282 non-null  float64       
 5   Item Class                        56993 non-null  object        
 6   Item Number                       65241 non-null  object        
 7   Item                              65282 non-null  object        
 8   Line Number                       65282 non-null  float64       
 9   List Price                        65282 non-null  float64       
 10  Order Number                      65282 non-nu

In [23]:
#Creating the new Dataframe by using necessary columns

final_df= df1[['CustKey', 'Item','Invoice Date','Invoice_year','Invoice_Month','Invoice_Date','Sales Quantity','Sales Amount','Sales Amount Based on List Price','Discount Amount','Sales Margin Amount','Sales Cost Amount','Sales Rep','U/M','List Price','Sales Price']]

In [24]:
final_df.isnull().sum()

CustKey                             0
Item                                0
Invoice Date                        0
Invoice_year                        0
Invoice_Month                       0
Invoice_Date                        0
Sales Quantity                      0
Sales Amount                        0
Sales Amount Based on List Price    0
Discount Amount                     2
Sales Margin Amount                 0
Sales Cost Amount                   0
Sales Rep                           0
U/M                                 0
List Price                          0
Sales Price                         1
dtype: int64

In [25]:
#Dropping the null values in Discount Amount and sales price has we have 60000+ data.

final_df.dropna(subset=['Discount Amount', 'Sales Price'], axis=0, inplace=True)

In [27]:
final_df.isnull().sum()

CustKey                             0
Item                                0
Invoice Date                        0
Invoice_year                        0
Invoice_Month                       0
Invoice_Date                        0
Sales Quantity                      0
Sales Amount                        0
Sales Amount Based on List Price    0
Discount Amount                     0
Sales Margin Amount                 0
Sales Cost Amount                   0
Sales Rep                           0
U/M                                 0
List Price                          0
Sales Price                         0
dtype: int64

In [32]:
final_df.head()

CustKey                             Item Invoice Date  Invoice_year  \
0  10000481.0                 Urban Large Eggs   2017-04-30          2017   
1  10002220.0               Moms Sliced Turkey   2017-07-14          2017   
2  10002220.0  Cutting Edge Foot-Long Hot Dogs   2017-10-17          2017   
3  10002489.0                         Kiwi Lox   2017-06-03          2017   
4  10004516.0             High Top Sweet Onion   2017-05-27          2017   

   Invoice_Month Invoice_Date  Sales Quantity  Sales Amount  \
0              4   2017-04-30             1.0        237.91   
1              7   2017-07-14             1.0        456.17   
2             10   2017-10-17             1.0        438.93   
3              6   2017-06-03             1.0        211.75   
4              5   2017-05-27           455.0      89248.66   

   Sales Amount Based on List Price  Discount Amount  Sales Margin Amount  \
0                              0.00          -237.91               237.91   
1                            824.96           368.79               456.17   
2                            548.66           109.73               438.93   
3                              0.00          -211.75               211.75   
4                         185876.60         96627.94             89248.66   

   Sales Cost Amount  Sales Rep U/M  List Price  Sales Price  
0                0.0      184.0  EA        0.00   237.910000  
1                0.0      127.0  EA      824.96   456.170000  
2                0.0      127.0  EA      548.66   438.930000  
3                0.0      160.0  EA        0.00   211.750000  
4                0.0      124.0  SE      408.52   196.150901

In [33]:
final_df.tail()

CustKey                          Item Invoice Date  Invoice_year  \
65277  10017638.0              High Top Oranges   2018-03-21          2018   
65278  10017638.0         Landslide White Sugar   2018-03-21          2018   
65279  10017638.0             Moms Potato Salad   2018-03-21          2018   
65280  10017638.0  Better Fancy Canned Sardines   2018-03-21          2018   
65281  10017638.0             Imagine Popsicles   2018-03-21          2018   

       Invoice_Month Invoice_Date  Sales Quantity  Sales Amount  \
65277              3   2018-03-21             9.0        569.90   
65278              3   2018-03-21             2.0        462.81   
65279              3   2018-03-21             8.0        987.20   
65280              3   2018-03-21            36.0      27297.51   
65281              3   2018-03-21            48.0      27582.02   

       Sales Amount Based on List Price  Discount Amount  Sales Margin Amount  \
65277                           1075.68           505.78               329.95   
65278                            873.56           410.75                39.26   
65279                           1863.36           876.16               413.20   
65280                          51524.28         24226.77             11108.61   
65281                          52061.28         24479.26             13347.80   

       Sales Cost Amount  Sales Rep U/M  List Price  Sales Price  
65277             239.95      180.0  EA      119.52    63.322222  
65278             423.55      180.0  EA      436.78   231.405000  
65279             574.00      180.0  EA      232.92   123.400000  
65280           16188.90      180.0  EA     1431.23   758.264167  
65281           14234.22      180.0  EA     1084.61   574.625417

In [35]:
final_df.shape

(65280, 16)

In [28]:
final_df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 65280 entries, 0 to 65281
Data columns (total 16 columns):
 #   Column                            Non-Null Count  Dtype         
---  ------                            --------------  -----         
 0   CustKey                           65280 non-null  float64       
 1   Item                              65280 non-null  object        
 2   Invoice Date                      65280 non-null  datetime64[ns]
 3   Invoice_year                      65280 non-null  int64         
 4   Invoice_Month                     65280 non-null  int64         
 5   Invoice_Date                      65280 non-null  object        
 6   Sales Quantity                    65280 non-null  float64       
 7   Sales Amount                      65280 non-null  float64       
 8   Sales Amount Based on List Price  65280 non-null  float64       
 9   Discount Amount                   65280 non-null  float64       
 10  Sales Margin Amount               65280 non-nu

In [31]:
final_df.describe().round(2)

CustKey  Invoice_year  Invoice_Month  Sales Quantity  Sales Amount  \
count     65280.00      65280.00       65280.00        65280.00      65280.00   
mean   10017702.67       2017.96           6.31           45.09       2852.04   
std        7176.24          0.95           3.56          429.67      15164.57   
min    10000453.00       2017.00           1.00            1.00        200.01   
25%    10012715.00       2017.00           3.00            2.00        308.38   
50%    10019665.00       2018.00           6.00            3.00        553.94   
75%    10023511.00       2019.00           9.00            8.00       1279.99   
max    10027583.00       2019.00          12.00        16000.00     555376.00   

       Sales Amount Based on List Price  Discount Amount  Sales Margin Amount  \
count                          65280.00         65280.00             65280.00   
mean                            4707.62          1855.57              1191.01   
std                            20696.74          9037.14              5860.86   
min                                0.00       -255820.80             -3932.93   
25%                              561.04           246.04               129.95   
50%                              998.16           441.76               246.49   
75%                             2316.63           999.76               579.39   
max                           632610.16        343532.66            188800.00   

       Sales Cost Amount  Sales Rep  List Price  Sales Price  
count           65280.00   65280.00    65280.00     65280.00  
mean             1661.03     137.42      514.71       283.70  
std              9556.63      26.64      449.19       252.03  
min                 0.00     103.00        0.00         0.34  
25%               167.79     113.00      181.56       100.07  
50%               304.53     134.00      325.19       183.76  
75%               687.40     160.00      803.86       448.22  
max            366576.00     185.00     2760.70      6035.00

In [37]:
final_df["Item"].value_counts()

Better Fancy Canned Sardines    1648
Ebony Prepared Salad            1471
Moms Sliced Turkey              1192
Imagine Popsicles               1191
Discover Manicotti              1126
                                ... 
Fast Salted Pretzels               1
Great Bagels                       1
Imagine Frozen Cauliflower         1
Jeffers Corn Puffs                 1
Bravo Fancy Canned Sardines        1
Name: Item, Length: 657, dtype: int64

Observation : The Top 5 Items sold were 
  1. Better Fancy Canned Sardiness
  2. Ebony prepared salad
  3. Moms Sliced Turkey
  4. Imagined Popsicles
  5. Discover Manicotti

In [39]:
final_df.to_excel("Pre_Processed_Amazon_Sales_Dataset.xlsx")